src/UI.ts

```diff
@@ -0,0 +1,102 @@
+import {
+  EmojiIdentifierResolvable,
+  MessageActionRow,
+  MessageButtonOptions,
+  MessageButtonStyleResolvable,
+} from 'discord.js';
+import { ButtonHandler } from './ButtonHandler';
+import { v4 as uuidv4 } from 'uuid';
+
+export type UIComponent = DispatchButton | DispatchLinkButton;
+
+export type ButtonOptions = {
```

Comment:
Can't this be an interface that extends `MessageButtonOptions`?

Reply:
Possibly, I've never used `extends` with an interface. I'll look into it

Reply:
Ok I did a thing with `Omit`, lmk what you think